pkg/sdn/plugin/subnets.go

```diff
@@ -95,11 +95,6 @@ func (master *OsdnMaster) deleteNode(nodeName string) error {
 	if err != nil {
 		return fmt.Errorf("Error fetching subnet for node %q for deletion: %v", nodeName, err)
 	}
-	_, ipnet, err := net.ParseCIDR(sub.Subnet)
-	if err != nil {
-		return fmt.Errorf("Error parsing subnet %q for node %q for deletion: %v", sub.Subnet, nodeName, err)
-	}
-	master.subnetAllocator.ReleaseNetwork(ipnet)
```

Comment:
HostSubnets aren't created synchronously when Nodes are created, they're created by the same Node watch that does HostSubnet deletion. So I think we're ok:

t=0, node exists, subnet exists
t=1, node is deleted
t=2, node is created
t=3, node deletion event is delivered, SDN master deletes the HostSubnet but does not yet release the IP range
t=4, node creation event is delivered, SDN master calls HostSubnets.Get(), sees that no HostSubnet exists for the node, and creates a new one, allocating a new IP range
t=5, HostSubnet deletion event is delivered, SDN master releases the (old) IP range, SDN nodes delete OVS rules for that node
t=6, HostSubnet creation event is delivered, SDN nodes add new OVS rules for the node

(Except that if you have used up your entire clusterNetworkCIDR, then this sequence of events would fail, because at t=4 there wouldn't be any IP subnets available yet. But in the non-racy case where you actually let the deletion get fully processed before recreating the node, it's fine.)